pkg/storaged/optional-panel.jsx

```diff
@@ -0,0 +1,256 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+
+import { show_modal_dialog } from "cockpit-components-dialog.jsx";
+import * as PK from "packagekit.es6";
+
+import { fmt_size, format_to_array, get_config } from "./utils.js";
+import { StorageButton } from "./storage-controls.jsx";
+
+const _ = cockpit.gettext;
+
+function install_dialog(data, action) {
+    var summary;
+    var extra_details = null;
+    var remove_details = null;
+
+    var missing_names = <strong>{data.missing_names.join(", ")}</strong>;
+
+    if (data.extra_names.length > 0 && data.remove_names.length == 0) {
+        summary = (
+            <p>
+                { format_to_array(_("$0 will be installed, along with additional packages:"), missing_names) }
```

Comment:
This should use ngettext. Depending on if it's one or more packages, translation will be different in non-English languages; e. g. in German, "a wird installiert" vs. "a, b werden installiert". This also occurs a few times below.